auth/simple_token.go

```diff
@@ -21,13 +21,87 @@ import (
 	"crypto/rand"
 	"math/big"
 	"strings"
+	"time"
 )
 
 const (
 	letters                  = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
 	defaultSimpleTokenLength = 16
+	simpleTokenTTL           = 5 * time.Minute
+	simpleTokenTTLResolution = 1 * time.Second
 )
 
+type simpleTokenTTLKeeper struct {
+	tokens              map[string]time.Time
+	addSimpleTokenCh    chan string
+	resetSimpleTokenCh  chan string
+	deleteSimpleTokenCh chan string
+	stopCh              chan struct{}
+	deleteTokenFunc     func(string)
+}
+
+func NewSimpleTokenTTLKeeper(deletefunc func(string)) *simpleTokenTTLKeeper {
+	stk := &simpleTokenTTLKeeper{
+		tokens:              make(map[string]time.Time),
+		addSimpleTokenCh:    make(chan string, 1),
+		resetSimpleTokenCh:  make(chan string, 1),
+		deleteSimpleTokenCh: make(chan string, 1),
+		stopCh:              make(chan struct{}),
+		deleteTokenFunc:     deletefunc,
+	}
+	return stk
+}
+
+func (tm *simpleTokenTTLKeeper) start() {
+	tm.stopCh = make(chan struct{})
+	go tm.run()
+}
+
+func (tm *simpleTokenTTLKeeper) stop() {
+	close(tm.stopCh)
+}
```

Comment:
This should wait for goroutine completion or it might look like a goroutine leak. Add `<-tm.donec` so `stop` won't exit until the goroutine is complete?